cmd/sourced/compose/workdir/workdir.go

```diff
@@ -0,0 +1,153 @@
+package workdir
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	goerrors "gopkg.in/src-d/go-errors.v1"
+
+	datadir "github.com/src-d/sourced-ce/cmd/sourced/dir"
+)
+
+const activeDir = "__active__"
+
+var (
+	// RequiredFiles list of required files in a directory to treat it as a working directory
+	RequiredFiles = []string{".env", "docker-compose.yml"}
+
+	// OptionalFiles list of optional files that could be deleted when pruning
+	OptionalFiles = []string{"docker-compose.override.yml"}
+
+	// ErrMalformed is the returned error when the workdir is wrong
+	ErrMalformed = goerrors.NewKind("workdir %s is not valid: %s")
+)
+
+type WorkdirType int
+
+const (
+	Local WorkdirType = iota
+	Orgs
+)
+
+type Workdir struct {
+	Type WorkdirType
+	Name string
+	Path string
+}
+
+type builder struct {
+	workdirsPath string
+}
+
+func (b *builder) build(path string) (*Workdir, error) {
+	wdType, err := b.typeFromPath(path)
+	if err != nil {
+		return nil, err
+	}
+
+	wdName, err := b.workdirName(*wdType, path)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Workdir{
+		Type: *wdType,
+		Name: wdName,
+		Path: path,
+	}, nil
+}
+
+func (b *builder) workdirName(wdType WorkdirType, path string) (string, error) {
+	var subPath string
+	switch wdType {
+	case Local:
+		subPath = "local"
+	case Orgs:
+		subPath = "orgs"
+	}
+
+	encoded, err := filepath.Rel(filepath.Join(b.workdirsPath, subPath), path)
+	if err != nil {
+		return "", err
+	}
+
+	decoded, err := base64.URLEncoding.DecodeString(encoded)
+	if err == nil {
+		return string(decoded), nil
+	}
+
+	return "", err
+}
+
+func (b *builder) typeFromPath(path string) (*WorkdirType, error) {
```

Comment:
better return WorkdirType here. And add type None.

Reply:
Done [here](https://github.com/src-d/sourced-ce/pull/183/commits/0bc94131c4f631543c32189807a46c0593078f8a).